Modules/posixmodule.c

```diff
@@ -16733,6 +16733,71 @@ INITFUNC(void)
     return PyModuleDef_Init(&posixmodule);
 }
 
+static PerfMapState perf_map_state;
```

Comment:
I don't think this is a good place for this API, given that the other public `trampoline` APIs are in the sys module:

https://docs.python.org/3.12/library/sys.html#sys.activate_stack_trampoline

Let's move this one to the sys module as well.